test/provider_os_memory_multiple_NUMA_nodes.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2024 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "base.hpp"
+
+#include "cpp_helpers.hpp"
+#include "provider_os_memory_internal.h"
+#include "umf/providers/provider_os_memory.h"
+#include <numa.h>
+#include <numaif.h>
+#include <umf/memory_provider.h>
+
+using umf_test::test;
+
+static umf_os_memory_provider_params_t UMF_OS_MEMORY_PROVIDER_PARAMS_TEST = {
+    /* .protection = */ UMF_PROTECTION_READ | UMF_PROTECTION_WRITE,
+    /* .visibility = */ UMF_VISIBILITY_PRIVATE,
+
+    // NUMA config
+    /* .nodemask = */ NULL,
+    /* .maxnode = */ 0,
+    /* .numa_mode = */ UMF_NUMA_MODE_BIND,
+    /* .numa_flags = */ 0,
+
+    // others
+    /* .traces = */ 1,
+};
+
+std::vector<std::tuple<int, int>> get_available_NUMA_nodes_numbers() {
+    int max_node = numa_max_node() + 1;
+    std::vector<std::tuple<int, int>> available_NUMA_nodes(max_node);
+
+    // Get all available NUMA nodes numbers
+    for (int i = 0; i <= max_node; ++i) {
+        available_NUMA_nodes.emplace_back(
+            i, numa_bitmask_isbitset(numa_all_nodes_ptr, i));
+    }
+
+    return available_NUMA_nodes;
+}
+
+std::tuple<int, int> get_numa_node(void *addr) {
+    int max_node = numa_max_node() + 1;
+    int numa_node;
+    int ret = get_mempolicy(&numa_node, NULL, max_node, addr,
```

Comment:
`max_node` can be `0` if you pass nullptr nodemask. Also, use nullptr instead of NULL in C++ code to differentiate between a pointer and the `0` value.

Reply:
Done.